SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestWorldComposition/SpatialTestWorldComposition.cpp

```diff
@@ -0,0 +1,192 @@
+		AddStepFromDefinition(ClientMovePawnStepDefinition, FWorkerDefinition::Client(1));
+
+		// Note that since initially, the pawn spawns close to the origin, it will load the ReplicatedActorLevel and ReplicatedAndNetLoadOnClientLevel, failing to unload those will first be seen in this step. 
+		// Test that the InitiallyDormantActorLevel was loaded correctly and the ReplicatedActorLevel and ReplicatedAndNetLoadOnClientLevel were unloaded.
+		AddStep(TEXT("SpatialTestWorldCompositionClientCheckTestLocation0"), FWorkerDefinition::Client(1), nullptr, nullptr, [this](ASpatialFunctionalTest* NetTest, float DeltaTime)
+			{
+				if (bIsCorrectAtLocation(0))
+				{
+					FinishStep();
+				}
+			}, 10.0f);
+
+		// Move the Pawn to the TestLocation with index 1.
+		AddStepFromDefinition(ClientMovePawnStepDefinition, FWorkerDefinition::Client(1));
+
+		// Test that the ReplicatedActorLevel was loaded correctly and the IntiallyDormantActorLevel was unloaded.
+		AddStep(TEXT("SpatialTestWorldCompositionClientCheckTestLocation1"), FWorkerDefinition::Client(1), nullptr, nullptr, [this](ASpatialFunctionalTest* NetTest, float DeltaTime)
+			{
+				if (bIsCorrectAtLocation(1))
+				{
+					FinishStep();
+				}
+			}, 10.0f);
+
+		// Move the Pawn to the TestLocation with index 2.
+		AddStepFromDefinition(ClientMovePawnStepDefinition, FWorkerDefinition::Client(1));
+
+		// Test that the ReplicatedAndNetLoadOnClientLevel was loaded correctly and the ReplicatedActorLevel was unloaded.
+		AddStep(TEXT("SpatialTestWorldCompositionClientCheckTestLocation2"), FWorkerDefinition::Client(1), nullptr, nullptr, [this](ASpatialFunctionalTest* NetTest, float DeltaTime)
+			{
+				if (bIsCorrectAtLocation(2))
+				{
+					FinishStep();
+				}
+			}, 10.0f);
+
+		// Move the Pawn to the TestLocation with index 3.
+		AddStepFromDefinition(ClientMovePawnStepDefinition, FWorkerDefinition::Client(1));
+
+		// Test that the InitiallyDormantAndNetLoadLevel was loaded correctly and the ReplicatedAndNetLoadOnClientLevel was unloaded.
+		AddStep(TEXT("SpatialTestWorldCompositionClientCheckTestLocation3"), FWorkerDefinition::Client(1), nullptr, nullptr, [this](ASpatialFunctionalTest* NetTest, float DeltaTime)
+			{
+				if (bIsCorrectAtLocation(3))
+				{
+					FinishStep();
+				}
+			}, 10.0f);
+
+		// Move the Pawn to the TestLocation with index 4.
+		AddStepFromDefinition(ClientMovePawnStepDefinition, FWorkerDefinition::Client(1));
+
+		// Test that both ReplicatedActorLevel and ReplicatedAndNetLoadOnClientLevel were loaded correctly, and that the InitiallyDormantAndNetLoadLevel was unloaded.
+		AddStep(TEXT("SpatialTestWorldCompositionClientCheckTestLocation4"), FWorkerDefinition::Client(1), nullptr, nullptr, [this](ASpatialFunctionalTest* NetTest, float DeltaTime)
+			{
+				if (bIsCorrectAtLocation(4))
+				{
+					FinishStep();
+				}
+			}, 10.0f);
+	}
+}
+
+bool ASpatialTestWorldComposition::bIsCorrectAtLocation(int TestLocation)
```

Comment:
Prefer to pass an array of position instead of a magic "int" number

Reply:
Not quite following what you mean by that, could you please be a little more specific?

Reply:
Oh, thank you for the clarification, will try to refactor this and also remove the b 😄!

Reply:
The function's signature should be something like : 
`struct ExpectedActor
{
  FVector Location;
  UClass* Class;
}

bool IsCorrectAtLocation(TArray<ExpectedActor> Expected)`

Reply:
A bit different from what I had in mind, but it works for me 👍

Reply:
Yeah I kind of wanted to make it so that if one wants to add another test location that contains multiple Actors it can be done as easily as possible. Might have complicated things a bit to be honest so not sure if it was a good idea after all?

Reply:
Personally I find it much easier to see what's happening in the test now, but that is probably super biased since I wrote it to begin with 😄